internal/services/watch_btc_events.go

```diff
@@ -334,6 +346,51 @@ func (s *Service) handleWithdrawal(
 	)
 }
 
+func (s *Service) handleUnexpectedUnbondingTx(
+	ctx context.Context,
+	spendingTx *wire.MsgTx,
+	spendingHeight uint32,
+	delegation *model.BTCDelegationDetails,
+) error {
+	registeredUnbondingTxBytes, parseErr := hex.DecodeString(delegation.UnbondingTx)
+	if parseErr != nil {
+		return fmt.Errorf("failed to decode unbonding tx: %w", parseErr)
+	}
+
+	registeredUnbondingTx, parseErr := bbn.NewBTCTxFromBytes(registeredUnbondingTxBytes)
+	if parseErr != nil {
+		return fmt.Errorf("failed to parse unbonding tx: %w", parseErr)
+	}
+
+	// This should never happen as we've already validated it's an unexpected tx
+	if registeredUnbondingTx.TxHash().String() == spendingTx.TxHash().String() {
+		return fmt.Errorf("inconsistent state: tx %s was marked as unexpected but matches registered unbonding tx",
+			spendingTx.TxHash().String())
+	}
+
+	log.Error().
+		Str("staking_tx", delegation.StakingTxHashHex).
+		Str("spending_tx", spendingTx.TxHash().String()).
+		Str("spending_height", strconv.FormatUint(uint64(spendingHeight), 10)).
+		Str("registered_unbonding_tx", registeredUnbondingTx.TxHash().String()).
+		Msg("detected unexpected unbonding transaction")
+
+	// Update delegation state to unbonding
+	subState := types.SubStateEarlyUnbonding
+	if err := s.db.UpdateBTCDelegationState(
+		ctx,
+		delegation.StakingTxHashHex,
+		types.QualifiedStatesForUnbondedEarly(),
+		types.StateUnbonding,
+		db.WithSubState(subState),
+		db.WithBtcHeight(int64(spendingHeight)),
+	); err != nil {
+		return fmt.Errorf("failed to update BTC delegation state: %w", err)
+	}
```

Comment:
why we only update the state for unexpected unbonding? We also need to update state in normal case right?